source/common/grpc/common.cc

```diff
@@ -269,5 +286,98 @@ std::string Common::typeUrl(const std::string& qualified_name) {
   return typeUrlPrefix() + "/" + qualified_name;
 }
 
+struct BufferInstanceContainer {
```

Comment:
I'm wondering could those ref-counted classes be std::shared_ptr?

Reply:
The problem is that the API provided by grpc::Slice supports storing a raw pointer and a function pointer.  A std::function is bigger than a raw pointer as is a std::shared_ptr andcasting them down to bit patterns isn't something I thought appropriate.  If you have a better way of doing this I am all ears as this is clunky.